### PR TITLE
Membership Registration

### PR DESCRIPTION
**Pull request by c00k133**
This PR adds an online registration form admin tools to our membership management system.

Here are the following additions:
- An online registration form under `/registration/`
    - This includes an automatic email notification after submission
- Admin page for seeing all applicants and their information
    - An admin may change anything in the applicants form
    - An admin may accept or decline an application
    - An admin may accept multiple applications at once

Once an application is accepted it will automatically be inserted into the member database and deleted.

Commit messages and comments are hopefully enough to convey design choices, please comment if not so.

*Illustrations:*

Registration page:
registration

Admin page:
admin

### CODE REVIEW

**Comment by c00k133** [comment]
Stuff to still (possibly) implement:
- [] Proper tests
- [] Automatic member status once accepted (i.e. gone to Namö)
- [] Analytics on languages

**Comment by c00k133** [comment]
The Phuxivator has given thumbs up so the PR goes out of WIP.

File: teknologr/api/views.py [hunk 1]
```diff
@@ -545,3 +611,48 @@ def arskDump(request):
...
+
+    dumpname = 'filename="regEmailDump_{}.csv"'.format(datetime.today().date())
+    return Response(
+            content,
+            status=200,
+            headers={'Content-Disposition': 'attachment; {}'.format(dumpname)}
+        )
+
+
+class ApplicantLanguages(csv_renderer.CSVRenderer):
```
Comment (tlangens) [inline teknologr/api/views.py]: `ApplicantLanguagesRenderer` would be more descriptive

File: teknologr/members/programmes.py [hunk 2]
```diff
@@ -1,32 +1,45 @@
...
-    'Elektronik och elektroteknik',
-    'Den byggda miljön',
-    'Energi- och miljöteknik',
-    'Maskin- och byggnadsteknik',
-    'Datateknik',
-    'Informationsnätverk',
-    'Produktionsekonomi',
-    'Teknisk fysik och matematik',
-)
+DEGREE_PROGRAMME_CHOICES = {
```
Comment (tlangens) [inline teknologr/members/programmes.py]: Glad to see someone finally updated this list

File: teknologr/registration/browsertests/test-registration.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.select import Select
+from selenium.webdriver.common.action_chains import ActionChains
+from registration.forms import format_programmes
+import string
+import random
+
+
+class RegistrationTest(LiveServerTestCase):
```
Comment (tlangens) [inline teknologr/registration/browsertests/test-registration.py]: Do my eyes deceive me? A test?!

File: teknologr/api/views.py (hunk 1 above)
Comment (c00k133) [reply teknologr/api/views.py]: The naming convention apparently slipped by me, fixed now.

File: teknologr/members/programmes.py (hunk 2 above)
Comment (c00k133) [reply teknologr/members/programmes.py]: Yeah, some structure was nice, and we had left out some programmes that have existed for a while :P

File: teknologr/registration/browsertests/test-registration.py (hunk 3 above)
Comment (c00k133) [reply teknologr/registration/browsertests/test-registration.py]: Got tired of filling in the form by hand, so some Selenium testing was a good addition.

**Comment by c00k133** [comment]
> I cannot immediately find anything terribly wrong. Very glad you put the registration in a separate app. The API views are very messy but they were before too.
> I'm sad to be reminded how messy this code is, and it's mostly my fault. Good to see somebody still managing to develop for it :)

The previous code helped a lot in developing this! 
I'll merge the PR now, and deploy it to production tomorrow.

File: teknologr/members/programmes.py (hunk 2 above)
Comment (tlangens) [reply teknologr/members/programmes.py]: They did not exist when the list was last updated (kandiuudistus was ongoing), which is why I deliberately made the field freetext editable.

**Pull request merged**